optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/incremental/IncrementalScoreDirector.java

```diff
@@ -42,6 +42,12 @@ public IncrementalScoreDirector(IncrementalScoreDirectorFactory scoreDirectorFac
                                     boolean constraintMatchEnabledPreference,
                                     IncrementalScoreCalculator<Solution_> incrementalScoreCalculator) {
         super(scoreDirectorFactory, constraintMatchEnabledPreference);
+        if (constraintMatchEnabledPreference
+                && !(incrementalScoreCalculator instanceof ConstraintMatchAwareIncrementalScoreCalculator)) {
+            throw new IllegalStateException("The score calculator " + incrementalScoreCalculator + " must implement "
+                    + ConstraintMatchAwareIncrementalScoreCalculator.class.getSimpleName()
+                    + " when constraintMatchEnabledPreference is true.");
+        }
```

Comment:
This causes the single test failure in optaplanner-examples. My intention was to fail fast if constraint matches are enabled (requested?) e.g. by one of the asserted modes. I am not sure if this is desired or not. Need to find out...